vm/asm.go

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2018 De-labtory
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package vm
+
+// Converts rawByteCode to assembly code.
+func assemble(rawByteCode []byte) (*asm, error) {
+	analysis()
+	return &asm{}, nil
+}
+
+// Do some analysis step (calculating the cost of running the code)
+func analysis() {
+
+}
+
+// Assembler has assembly codes and the current program counter(position)
+type assembler interface {
+	next() Hexer
+	jump(pc uint64)
+}
+
+type Hexer interface {
+	hex() []uint8
+}
+
+type asm struct {
```

Comment:
The instruction is like `opcode + data`. But some are `opcodes` and the others are `data`.  
It would be nice to have a word that can express both opcode and data.
I will think about it more!